File: library/src/main/java/org/meganekkovr/animation/QuaternionEvaluator.java
package org.meganekkovr.animation;

import android.animation.TypeEvaluator;

import org.joml.Quaternionf;

/**
 * This evaluator can be used to perform type interpolation between {@link Quaternionf} values.
 */
class QuaternionEvaluator implements TypeEvaluator<Quaternionf> {
    private final Quaternionf mResult = new Quaternionf();

    @Override
    public Quaternionf evaluate(float fraction, Quaternionf startValue, Quaternionf endValue) {
        return startValue.slerp(endValue, fraction, mResult);
    }
}

File: library/src/main/java/org/meganekkovr/animation/QuaternionEvaluator.kt
package org.meganekkovr.animation

import android.animation.TypeEvaluator

import org.joml.Quaternionf

/**
 * This evaluator can be used to perform type interpolation between [Quaternionf] values.
 */
internal class QuaternionEvaluator : TypeEvaluator<Quaternionf> {
    private val mResult = Quaternionf()

    override fun evaluate(fraction: Float, startValue: Quaternionf, endValue: Quaternionf): Quaternionf {
        return startValue.slerp(endValue, fraction, mResult)
    }
}
